src/remote-data.ts

```diff
@@ -787,30 +815,43 @@ const ap = <L, A, B>(fab: RemoteData<L, Function1<A, B>>, fa: RemoteData<L, A>):
 const map = <L, A, B>(fa: RemoteData<L, A>, f: Function1<A, B>): RemoteData<L, B> => fa.map(f);
 const chain = <L, A, B>(fa: RemoteData<L, A>, f: Function1<A, RemoteData<L, B>>): RemoteData<L, B> => fa.chain(f);
 
-//Foldable
+//Foldable2v
 const reduce = <L, A, B>(fa: RemoteData<L, A>, b: B, f: Function2<B, A, B>): B => fa.reduce(f, b);
+const foldMap = <M>(M: Monoid<M>) => <L, A>(fa: RemoteData<L, A>, f: (a: A) => M): M =>
+	fa.isSuccess() ? f(fa.value) : M.empty;
+const foldr = <L, A, B>(fa: RemoteData<L, A>, b: B, f: (a: A, b: B) => B): B => (fa.isSuccess() ? f(fa.value, b) : b);
+
+//Traversable2v
+const traverse = <F>(F: Applicative<F>) => <L, A, B>(
+	ta: RemoteData<L, A>,
+	f: (a: A) => HKT<F, B>,
+): HKT<F, RemoteData<L, B>> => {
+	if (ta.isSuccess()) {
+		return F.map<B, RemoteData<L, B>>(f(ta.value), of);
+	} else if (ta.isFailure()) {
```

Comment:
Is it necessary to cover all `else` branches? Looks like they can be combined.

Reply:
The only reason I did this was so that I could be more explicit with the types. I might be able to do something like this:

```ts
else {
  return (F.of(ta) as unknown)
    as RemoteFailure<L, A> | as RemotePending<L, A> | as RemoteInitial<L, A>
}
```

I can try this later tonight.

Reply:
This worked. I agree this is better.